src/forklift/core.py

```diff
@@ -184,9 +184,13 @@ def abstract_type(type):
             if abstract_type(source_fld.type) != abstract_type(destination_fld.type):
                 mismatching_fields.append('{}: source type of {} does not match destination type of {}'
                                           .format(source_fld.name, source_fld.type, destination_fld.type))
-            elif source_fld.type == 'String' and source_fld.length != destination_fld.length:
-                mismatching_fields.append('{}: source length of {} does not match destination length of {}'
-                                          .format(source_fld.name, source_fld.length, destination_fld.length))
+            elif source_fld.type == 'String':
+                if source_fld.length > 4000:
+                    log.warn('%s is longer than 4000 characters. Truncation may occur.', source_fld.name)
+                    source_fld.length = 4000
+                if source_fld.length != destination_fld.length:
```

Comment:
Given

```py
#: setup test
source_fld.length = 5000
source_fld.type = 'String'
destination_fld.length = 4001

#: back to code in question
if source_fld.length > 4000:
    log.warn('%s is longer than 4000 characters. Truncation may occur.', source_fld.name)
    source_fld.length = 4000
if source_fld.length != destination_fld.length:
   mismatching_fields.append('{}: source length of {} does not match destination length of {}'
                                               .format(source_fld.name, source_fld.length, destination_fld.length))
```
both if statements will be true and a `ValidationException` will be thrown.

Am I missing an assumption that destination_fld.length will never be greater than 4000?